server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -150,6 +183,13 @@ private float parseToFloat(Object value) {
                 return result;
             }
 
+            @Override
+            public BiFunction<Script, ScriptCompiler, MapperScript> compiler(String fieldName) {
+                return (s, ss) -> {
+                    throw new IllegalArgumentException("Unknown parameter [script] for mapper [" + fieldName + "]");
```

Comment:
It's a user error - we share an implementation for all the numeric types, so this can happen if somebody tries to add a mapping `{ "type" : "int", "script" : "blah" }`